src/main/java/org/codeforamerica/shiba/SessionLogFilter.java

```diff
@@ -60,4 +67,10 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
     chain.doFilter(request, response);
     MDC.clear();
   }
+
+  private String createRequestIp(HttpServletRequest request) {
+    String requestIpHeader = Optional.ofNullable(request.getHeader("X-FORWARDED-FOR")).orElse("");
+    String[] ipAddresses = requestIpHeader.split(",");
+    return ipAddresses.length > 1 ? ipAddresses[ipAddresses.length - 2].trim() : "<blank>";
+  }
```

Comment:
C/P from FileDownloadController when we _were_ using it for DownloadCAF. This is now being used for logging ip